### ISSUE TITLE
[dst] initialization order (Trac #1664)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1656]
Initialize variables in the order they appear in the header.

```text
In file included from /home/dschultz/Documents/combo/trunk/src/dst/private/dst13/I3DSTDAQModule13.cxx:5:0:
    /home/dschultz/Documents/combo/trunk/src/dst/public/dst/dst13/I3DSTDAQModule13.
    h: In constructor ‘I3DSTDAQModule13::I3DSTDAQModule13(const I3Context
        &)’:
/home/dschultz/Documents/combo/trunk/src/dst/public/dst/dst13/I3DSTDAQModule13.h:77:8:
    warning: ‘I3DSTDAQModule13::dstHeaderWritten_’ will be initialized
    after [-Wreorder]
bool dstHeaderWritten_;
^~~~~~~~~~~~~~~~~
/home/dschultz/Documents/combo/trunk/src/dst/public/dst/dst13/I3DSTDAQModule13.h:63:7:
    warning:   ‘int I3DSTDAQModule13::eventIndex_’ [-Wreorder]
int eventIndex_;
^~~~~~~~~~~
/home/dschultz/Documents/combo/trunk/src/dst/private/dst13/I3DSTDAQModule13.cxx:29:1:
    warning:   when initialized here [-Wreorder]
I3DSTDAQModule13::I3DSTDAQModule13(const I3Context & ctx) :
^~~~~~~~~~~~~~~~
```

Migrated from https://code.icecube.wisc.edu/ticket/1664
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:12:47", 
    "description": "Initialize variables in the order they appear in the header.\n\n{{{\nIn file included from /home/dschultz/Documents/combo/trunk/src/dst/private/dst13/I3DSTDAQModule13.cxx:5:0:\n    /home/dschultz/Documents/combo/trunk/src/dst/public/dst/dst13/I3DSTDAQModule13.\n    h: In constructor \u2018I3DSTDAQModule13::I3DSTDAQModule13(const I3Context\n        &)\u2019:\n/home/dschultz/Documents/combo/trunk/src/dst/public/dst/dst13/I3DSTDAQModule13.h:77:8:\n    warning: \u2018I3DSTDAQModule13::dstHeaderWritten_\u2019 will be initialized\n    after [-Wreorder]\nbool dstHeaderWritten_;\n^~~~~~~~~~~~~~~~~\n/home/dschultz/Documents/combo/trunk/src/dst/public/dst/dst13/I3DSTDAQModule13.h:63:7:\n    warning:   \u2018int I3DSTDAQModule13::eventIndex_\u2019 [-Wreorder]\nint eventIndex_;\n^~~~~~~~~~~\n/home/dschultz/Documents/combo/trunk/src/dst/private/dst13/I3DSTDAQModule13.cxx:29:1:\n    warning:   when initialized here [-Wreorder]\nI3DSTDAQModule13::I3DSTDAQModule13(const I3Context & ctx) :\n^~~~~~~~~~~~~~~~\n}}}", 
    "reporter": "david.schultz", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067167842669", 
    "component": "combo reconstruction", 
    "summary": "[dst] initialization order", 
    "priority": "minor", 
    "keywords": "", 
    "time": "2016-04-26T20:47:17", 
    "milestone": "", 
    "owner": "juancarlos", 
    "type": "cleanup"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160725T20:21:21: 
* juancarlos changed status from "new" to "closed"
* juancarlos changed resolution from "" to "fixed"
* juancarlos commented:

> This problem was resolved with commit 148743 as part of ticket #1577.

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:12:47: 
* olivas changed milestone from "June 2016" to ""
* olivas commented:

> Milestone June 2016 deleted
